pkgs/top-level/all-packages.nix

```diff
@@ -10080,6 +10080,9 @@ with pkgs;
 
   importNpmLock = callPackages ../build-support/node/import-npm-lock { };
 
+  inherit (callPackages ../build-support/node/fetch-pnpm-deps { })
+    mkFetchPnpmDeps;
```

Comment:
> This seems a bit weird.. Why not `fetchPnpmDeps = pnpm.fetchDeps`? and/or `pnpmConfigureHook = pnpm.configHook`?

This still bothers me BTW...

Reply:
> I don't think we should have a top-level function for this. It is very possible that future iterations of this function _will_ break existing packages, so we shouldn't encourage people to use a top-level function that might be outdated after some time.

OK I agree with that, but I don't understand why would someone want to use `mkFetchPnpmDeps`, and not `pnpm_{8,9}.fetchDeps`?

Reply:
> mkFetchPnpmDeps is probably not very useful in the top-level, yes. I wasn't sure where to put it, because it seems like build helpers usually go into `pkgs/build-support`, but I didn't think `pnpm` itself should be in there?
> 
> So either I move the `fetchDeps` function into `pkgs/development/tools/pnpm` or I move `pnpm` alongside `fetchDeps` to `pkgs/build-support/node/pnpm`

Oh I think I understand, you import `mkFetchPnpmDeps` in `pnpm/generic.nix`... I think the simplest thing would be to not even define `mkFetchPnpmDeps`, and put the contents of this function in the `passthru` of `pnpm/generic.nix`.

Reply:
You could also define the `mkFetchPnpmDeps` function in a separate file near `pnpm/generic.nix` IMO.

Reply:
I still oposing exposing `mkFetchPnpmDeps`.